### ISSUE TITLE
port back to CMake

### ISSUE THREAD

**Issue by schnitzeltony** [liri-archive/wayland-archived#47]
If I am not mistaken this is the last package that needs backport for building liri-shell in qbs-less environment (e.g Openembedded)

**Comment by n3rdopolis** [comment]
It seems that I can't build it, even with qbs anymore... ...I changed the build scripts for say QtUdev... ...but qbs no longer finds it...


```
ERROR: /srcbuild/lirios-wayland/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_support/eglfs_kms_support.qbs:27:5 Error while handling product 'LiriEglFSKmsSupport':
/srcbuild/lirios-wayland/src/plugins/platforms/eglfs/eglfsdeviceintegration.qbs:111:13 Dependency 'Qt5Udev' not found for product 'LiriEglFSKmsSupport'.
/srcbuild/lirios-wayland/src/plugins/platforms/eglfs/eglfs-plugin.qbs:3:1 Error while handling product 'lirieglfs':
/srcbuild/lirios-wayland/src/plugins/platforms/eglfs/eglfsdeviceintegration.qbs:111:13 Dependency 'Qt5Udev' not found for product 'lirieglfs'.
/srcbuild/lirios-wayland/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms_egldevice/eglfs_kms_egldevice.qbs:3:1 Error while handling product 'eglfs-kms-egldevice-integration':
/srcbuild/lirios-wayland/src/plugins/platforms/eglfs/eglfsdeviceintegration.qbs:111:13 Dependency 'Qt5Udev' not found for product 'eglfs-kms-egldevice-integration'.
/srcbuild/lirios-wayland/tests/manual/libinput/libinput.qbs:3:1 Error while handling product 'tst_wl_libinput':
/srcbuild/lirios-wayland/src/libinput/libinput.qbs:77:13 Dependency 'Qt5Udev' not found for product 'tst_wl_libinput'.
/opt/share/qbs/imports/LiriModuleProject.qbs:82:5 Error while handling product 'LiriEglFSKmsSupport-pkgconfig':
/srcbuild/lirios-wayland/src/plugins/platforms/eglfs/eglfsdeviceintegration.qbs:111:13 Dependency 'Qt5Udev' not found for product 'LiriEglFSKmsSupport-pkgconfig'.
/srcbuild/lirios-wayland/src/plugins/platforms/eglfs/deviceintegration/eglfs_kms/eglfs_kms.qbs:3:1 Error while handling product 'eglfs-kms-integration':
/srcbuild/lirios-wayland/src/plugins/platforms/eglfs/eglfsdeviceintegration.qbs:111:13 Dependency 'Qt5Udev' not found for product 'eglfs-kms-integration'.
/opt/share/qbs/imports/LiriModuleProject.qbs:145:5 Error while handling product 'LiriEglFSKmsSupport-cmake':
/srcbuild/lirios-wayland/src/plugins/platforms/eglfs/eglfsdeviceintegration.qbs:111:13 Dependency 'Qt5Udev' not found for product 'LiriEglFSKmsSupport-cmake'.
/srcbuild/lirios-wayland/src/plugins/platforms/eglfs/eglfsdeviceintegration.qbs:33:5 Error while handling product 'LiriEglFSDeviceIntegration':
/srcbuild/lirios-wayland/src/plugins/platforms/eglfs/eglfsdeviceintegration.qbs:48:9 Dependency 'Qt5Udev' not found for product 'LiriEglFSDeviceIntegration'.

```

**Comment by schnitzeltony** [comment]
Turns out that liri-shell builds happily without this 'wayland', So I could close this one - **but**:

@plfiorini Might turn off-topic but: I lost liri by the decision to use qbs. It is not that I did not invest time in getting qbs flying on openembedded [1] but I did not find the right hooks to keep qbs from changing all build flags..

Now things changed: We are back to cmake and again I want to get Liri running on at least a Raspi. 
Think I have all liri packets required in an image but...  - and now comes the 'fun ' part:
sddm seems to have problems with all sessions starting with 'L'. (the 'L' not important but fun) Have already helped LxQt and Lumina with [2] (sddm is a bit over-enthusiastic when it comes to TryExec) but now it is liri. sddm does not add liri to greeter's combobox.
To be continued...

[1] https://github.com/schnitzeltony/meta-qt5/tree/rocko-qbs
[2] https://github.com/schnitzeltony/meta-qt5-extra/blob/master/recipes-misc/recipes-graphics/sddm/sddm/0002-Workaround-missing-sessions.patch

**Comment by schnitzeltony** [comment]
should add some more background on sddm patch [1]:

if I remember correctly: 
* desktop-files not containing 'TryExec' were ignored
* path-less entries at TryExec were not displayed: A set 'PATH' in systemd service environment is nothing to rely on in a default environment. So patch ignores path-less 

Still the question why there is no liri...

[1] https://github.com/schnitzeltony/meta-qt5-extra/blob/master/recipes-misc/recipes-graphics/sddm/sddm/0002-Workaround-missing-sessions.patch

**Comment by plfiorini** [comment]
liri-wayland has been split in various pieces: fullscreen-shell is upstreamed and will be available with Qt 5.13, materialdecoration has its own repository but it will be deprecated soon, some code went into liri-shell and the QPA part is in liri-eglfs.

**Comment by plfiorini** [comment]
Closing all the old issues, new code is about to be force pushed.
